### DISCUSSION TITLE
ELSA 3 - Get Workflow Definition by URL Parameter

### DISCUSSION THREAD

**Discussion by hpposch** [elsa-workflows/elsa-core#4102] [Q&A]
Hi all.
In Elsa 2 you have been able to query the workflow designer for a specific definition via a url parameter.
For example: http://localhost:5163/workflow-definitions/21fa1858e2ed40e5adc93597c49ffc78
Is there anything like that in elsa 3?
Or is there another way (http header or something) to call the correct designer?

Thanks in advance for your help

**Comment by MikeDLi** [comment]
Hi-

This feature is needed when embedding the designer in an external application:
1) The external application can request an access token via Api (already available in v3-rc1)
2) Then sends a request such as: https://localhost:5163/workflow-definitions/xxxx?token={access_token}

Thanks

**Comment by sfmskywalker** [comment]
Hi @hpposch,

Thank you for bringing up this query. As of now, the Stencil implementation in Elsa 3 doesn't directly support querying the workflow designer via a URL parameter in the same way Elsa 2 did. However, as you might be interested to know, [Elsa Studio](https://github.com/elsa-workflows/elsa-studio) is the successor to the Elsa 3 Stencil implementation and offers this capability, similar to the Elsa 2 dashboard.
